src/Native/System.Native/pal_console.cpp

```diff
@@ -35,9 +35,10 @@ extern "C" int32_t GetWindowSize(WinSize* windowSize)
 #endif
 }
 
-extern "C" int32_t IsATty(int fd)
+extern "C" int32_t IsATty(intptr_t fd)
 {
-    return isatty(fd);
+    assert(INT32_MIN <= fd && fd <= INT32_MAX);
```

Comment:
I did a quick look, but I don't see a good define for this on Linux.  I changed it to use OPEN_MAX when it is defined.